language/move-vm/types/src/values/values_impl.rs

```diff
@@ -549,6 +579,20 @@ impl IndexedRef {
                 r1.borrow()[self.idx] == *r2.borrow()[other.idx].as_value_ref::<u8>()?
             }
 
+            (Locals(r1), VecU16(r2)) | (Struct(r1), VecU16(r2)) => {
```

Comment:
cc @runtian-zhou this is gonna conflict hard with your runtime checks

Reply:
It's fine. It's already lots of rebase!